app/controllers/scottish_postcodes_controller.js

```diff
@@ -1,20 +1,28 @@
 "use strict";
 
 const ScottishPostcode = require("../models/scottish_postcode");
-const Postcode = require("postcode");
+const Postcode = require("../models/postcode");
+const Pc = require("postcode");
 const {
   InvalidPostcodeError,
   PostcodeNotFoundError,
+  PostcodeNotScottishError,
 } = require("../lib/errors.js");
 
 exports.show = (request, response, next) => {
   const { postcode } = request.params;
-  if (!Postcode.isValid(postcode.trim()))
+  if (!Pc.isValid(postcode.trim()))
     return next(new InvalidPostcodeError());
 
   ScottishPostcode.find(postcode, (error, result) => {
     if (error) return next(error);
-    if (!result) return next(new PostcodeNotFoundError());
+    if (!result) {
```

Comment:
Can `next()` be called twice here? Once at LOC30 and once at LOC21-23

Reply:
Good catch - will `return` the value of `Postcode.find` to fix.

Reply:
Fixed I think - I'm not a Node.js expert, but I reckon that's going to work how I think it is.